public/lessons/greedy/exchange-arguments.mdx

```diff
@@ -0,0 +1,47 @@
+# Exchange Argument
+An exchange argument is a technique used to prove the optimality of greedy algorithms. It involves showing that any non-optimal solution can be transformed into an optimal one by exchanging elements in a strategic manner.
+
+### Steps to Apply Exchange Arguments
+1. Identify Greedy Choices: Identify the key decisions made by the greedy algorithm at each step.
+
+2. Define Exchange Strategy: Define a strategy for exchanging elements to transform any suboptimal solution into an optimal one.
+
+3. Prove Optimality: Use mathematical induction or contradiction to prove that the exchange strategy always leads to an optimal solution.
+
+## Example Problem
+Let's consider a classic example where exchange arguments can be applied: the Activity Selection Problem.
+
+### Activity Selection Problem
+Given a set of activities each marked by a start time and finish time, the goal is to select the maximum number of non-overlapping activities.
+
+### Greedy Algorithm
+1. Sort activities by finish time.
+
+2. Select the first activity.
+
+3. For each remaining activity, if its start time is after the finish time of the last selected activity, select it.
+
+### Exchange Argument
+1. Identify Greedy Choices: The greedy choice is to always select the activity with the earliest finish time that doesn't conflict with previously selected activities.
+
+2. Define Exchange Strategy: Suppose there exists an optimal solution where the greedy algorithm fails. We can exchange the conflicting activity in this solution with the one selected by the greedy algorithm. This exchange will not decrease the number of selected activities, and it might even increase it, contradicting the optimality of the assumed solution.
```

Comment:
Explain why this might increase it

Reply:
Also I feel like this bit feels like it is encroaching on "3"

Reply:
This is kinda why I gave the example of sorting by start times being wrong to help explain this, not sure how to better say that here?

Reply:
Something along the lines of "When performing this exchange, this could create more available time for an additional activity to be added in."